src/vendor/components/navigation/DurationPicker.jsx

```diff
@@ -0,0 +1,146 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { withStyles } from '@material-ui/core/styles';
+import InputLabel from '@material-ui/core/InputLabel';
+import MuiSlider from '@material-ui/lab/Slider';
+import { ArrayWrapper, selectFrom } from '../../vectors';
+import { coalesce, toArray } from '../../utils';
+
+const QUERY_TIME_FORMAT = 'yyyy-MM-dd';
+const styles = () => ({
+  root: {
+    display: 'inline-block',
+    verticalAlign: 'bottom',
+    position: 'relative',
+    paddingTop: '2rem',
+    margin: '0 20px',
+  },
+  label: {
+    display: 'block',
+    whiteSpace: 'nowrap',
+    padding: '0 0 0.5rem 0',
+    fontSize: '0.8rem',
+    color: '#757575',
+    textAlign: 'left',
+  },
+  value: {
+    display: 'block',
+    whiteSpace: 'nowrap',
+    padding: '0 10px 0.5rem 0',
+    fontSize: '1rem',
+    color: '#444444',
+    textAlign: 'left',
+  },
+  slider: {
+    display: 'block',
+    padding: '1rem 0 0 0',
+    height: '2rem',
+    position: 'absolute',
+    bottom: 0,
+  },
+});
+
+class DurationPickerPre extends Component {
+  constructor(props) {
+    super(props);
+
+    const { defaultValue, value } = props;
+    const width = coalesce(props.width, '8rem');
+    const options = toArray(props.options);
+    const range = { min: 0, max: options.length - 1, step: 1 };
+    const requestedIndex = options.findIndex(({ id }) => id === value);
+
+    if (requestedIndex !== -1) {
+      this.state = {
+        index: requestedIndex,
+        range,
+        options,
+        event: { target: {} },
+        width,
+      };
+
+      return;
+    }
+
+    const index = options.findIndex(({ id }) => id === defaultValue);
+
+    this.state = { index, range, options, event: { target: {} }, width };
+    this.handleDragEnd();
```

Comment:
This method will end up callling `this.setState` which we don't want in the constructor. `this.setState` causes a re-render of the component and all of its children. We don't want that in the constructor because the component hasn't rendered yet.